azure/batch/scripts/predict_on_catalog.py

```diff
@@ -104,39 +106,107 @@ def setup(self, stage: Optional[str] = None):
 
 def save_predictions_to_json(predictions, id_str, label_cols, save_loc):
     # JSON output format is used for services like the zooniverse subject assistant
+    # Could add any other decision rules into this function
     assert save_loc.endswith('.json')
-    # setup the output data structur with a schema describing the data
+    # setup the output data structure with a schema describing the data
     output_data = {
       'schema': {
         'version': 1,
         'type': 'zooniverse/subject_assistant',
-        'data': {'subject_id': 'probability_galaxy_is_not_smooth'}
+        'data': {
+            'predictions': {'subject_id': ['smooth-or-featured-cd_smooth', 'smooth-or-featured-cd_featured-or-disk', 'smooth-or-featured-cd_problem']},
+            'probabilities': {'subject_id': 'probability_at_least_20pc_featured'}
+        }
       }
+      # will add the actual data under 'data' key, below
     }
+
+    # check that the predictions mean what we think they mean
+    assert label_cols[0] == 'smooth-or-featured-cd_smooth'
+    assert label_cols[1] =='smooth-or-featured-cd_featured-or-disk'
+    assert label_cols[2] == 'smooth-or-featured-cd_problem'
+    # okay, now it's safe to hardcode the values below
+    
     # only derive each galaxies smooth or features question right now for simplicity of metric
     # i.e. we're trying to figure out if this galaxy is interesting or not for human volunteers
-    # if it's smooth it's not interesting so we can use this metric to decide to show it to volunteers
+    # if it's not featured it's not interesting so we can use this metric to decide to show it to volunteers
     smooth_or_featured_start_and_end_indices = [0, 2]
-    # the smooth answer label index
-    smooth_or_featured_smooth_index = 0
-    # lower bound of volunteers answering for a feature, i.e. 20% of volunteers say yes to the feature
-    odds_bound = 0.2
-    #
-    # variances = predictions_to_variance_of_answer(predictions, smooth_or_featured_start_and_end_indices, smooth_or_featured_smooth_index)
-    # expectations = predictions_to_expectation_of_answer(predictions, smooth_or_featured_start_and_end_indices, smooth_or_featured_smooth_index)
-    probability_galaxy_is_not_smooth = odds_answer_below_bounds(predictions,
+
+    # the featured answer label index
+    smooth_or_featured_featured_index = 1
+
+    # upper bound of volunteers answering for a feature, i.e. no more than e.g. 20% of volunteers select the answer (here, featured)
```

Comment:
Not blocking - I don't quite understand this comment in relation to the `odds_answer_below_bounds` function details but that's probably just me being thick.

> e.g. the predicted odds that an infinite number of volunteers would answer `smooth' to `smooth or featured' less than 20% of the time

So we're we're looking for the odds that at most 20% of the volunteers would say it's  'featured' less than 20% of the time - why 20%?

I assume this means that if the odds are high that any volunteer would say it's `featured` less than 20% of the time then the galaxy image is less likely to actually  have a feature in it?

any clarifying comment or answer here may be useful for folks in the future.